### ISSUE TITLE
Does this work with wdio v5?

### ISSUE THREAD

**Issue by domoritz** [LeadPages/wdio-static-server-service#13]
I'm working on updating a wdio4 setup to wdio5 and it's using the static server. When I run it I get an error that a globally defined function (in the index.html that is loaded by the static server) is not defined. Could the problem be that static server does not work with wdio5? What do I have to change to make it work?

You can find the config and the whole setup at https://github.com/vega/vega-lite/pull/4465.

**Comment by chris-arnold** [comment]
Unfortunately, this is no longer maintained since we migrating to using Chrome headless for our end-to-end testing.  I'll update our README to reflect this. If you end up digging into this any further, PRs are always welcome.

**Comment by domoritz** [comment]
Ahh, I see. Thanks for the note. Is there a migration path to chrome headless if I want to use an index.html page in my tests?

**Comment by chris-arnold** [comment]
There's a handful of solid examples for migrating to puppeteer out there, like [this one from GoDaddy](https://godaddy.github.io/2018/05/07/moving-from-webdriver-to-puppeteer/).

In regards to your question on loading functions from index.html ([I  assume you mean this?](https://github.com/vega/vega-lite/blob/master/test-runtime/index.html)), you should be able to do that from the puppeteer framework itself using page evaluate - here's some probably-not-exactly-correct sample code. (I'm a Python/backend dev, but a bit familiar with this area). 😄 
```javascript
await page.evaluate(() => {
  window.helperFunctions = {
    embed: function(vlSpec) {
      const vgSpec = vl.compile(vlSpec).spec;
      view = new vega.View(vega.parse(vgSpec))
        .renderer('svg')
        .initialize('#vis')
        .run();
    },
    mark: function(id, parent) {
      return document.querySelector((parent ? `g.${parent} ` : '') +
        `g.mark-symbol.role-mark path:nth-child(${id})`);
    }
}
```

**Comment by domoritz** [comment]
@chris-arnold Thank you for the code snippet. Am I understanding correctly that you are suggesting to drop wdio completely and switch to just use puppeteer? Do you have any suggested tutorials/docs for how to set this up with jest/mocha?

**Comment by chris-arnold** [comment]
It depends on how deeply entrenched you are in webdriver, really. 

We migrated much of our frontends to React about the same time we migrated to Jest, because we used it with mocking to render components for unit testing. It felt natural to extend into Puppeteer with Jest for the end-to-end tests, and we eventually refactored everything along those lines. Our test suites that use puppeteer ended up being fairly straightforward too - we started in the same place as that page you linked, and haven't strayed far from that pattern. (We have a few helper ES6 modules we import here and there to to handle parameters/environment/mocking/etc...)

We're using Chrome Headless in production as well, for taking screenshots on the backend of templates used to build landing pages. We run it in Kubernetes with our own API fronting it, and as long as we don't try and run it at a big scale, Chrome Headless has been really solid (way, way more solid than phantomjs in recent years anyway).

In short, I'm not so much suggesting you switch, but I am saying we ended up switching, and it went well, without having to deviate from standard patterns.

**Comment by domoritz** [comment]
Thanks for the insights. We already use jest and I am not happy with all the dependencies we pull in for the runtime tests. Switching to just Jest + Puppeteer sounds like a good option to me.

**Comment by domoritz** [comment]
This was super helpful. I switched to Puppeteer in https://github.com/vega/vega-lite/pull/4465.
